internal/core/command/get_test.go

```diff
@@ -108,9 +116,145 @@ func TestNewGetCommandNoCorrelationIDInContext(t *testing.T) {
+			[]string{"Content-MD5", "If-Modified-Since", "Max-Forwards"},
+			map[string]string{
+				"Content-MD5":       "",
+				"If-Modified-Since": "",
+				"Max-Forwards":      "",
+			},
+			newRequestWithHeaders(map[string]string{
+				"Content-MD5":       "Content-MD5",
+				"If-Modified-Since": "If-Modified-Since",
+				"Max-Forwards":      "Max-Forwards",
+			}, http.MethodGet),
+		},
+		{
+			"json content type propagated, content-md5 header not propagated, if-modified-since not propagated, max-forwards not propagated",
+			[]string{clients.ContentType, "Content-MD5", "If-Modified-Since", "Max-Forwards"},
+			map[string]string{
+				clients.ContentType: clients.ContentTypeJSON,
+				"Content-MD5":       "",
+				"If-Modified-Since": "",
+				"Max-Forwards":      "",
+			},
+			newRequestWithHeaders(map[string]string{
+				clients.ContentType: clients.ContentTypeJSON,
+				"Content-MD5":       "Content-MD5",
+				"If-Modified-Since": "If-Modified-Since",
+				"Max-Forwards":      "Max-Forwards",
+			}, http.MethodGet),
+		},
+		{
+			"cbor content type propagated, user-agent header not propagated, if-modified-since not propagated, max-forwards not propagated",
+			[]string{clients.ContentType, "User-Agent", "Content-MD5", "If-Modified-Since", "Max-Forwards"},
+			map[string]string{
+				clients.ContentType: clients.ContentTypeCBOR,
+				"User-Agent":        "",
+				"Content-MD5":       "",
+				"If-Modified-Since": "",
+				"Max-Forwards":      "",
+			},
+			newRequestWithHeaders(map[string]string{
+				clients.ContentType: clients.ContentTypeCBOR,
+				"User-Agent":        "User-Agent",
+				"Content-MD5":       "Content-MD5",
+				"If-Modified-Since": "If-Modified-Since",
+				"Max-Forwards":      "Max-Forwards",
+			}, http.MethodGet),
+		},
+		{
+			"json content type propagated, if-modified-since not propagated, content-encoding not propagated, cookie not propagated, max-forwards not propagated",
+			[]string{clients.ContentType, "If-Modified-Since", "Content-Encoding", "Cookie", "Max-Forwards"},
+			map[string]string{
+				clients.ContentType: clients.ContentTypeJSON,
+				"If-Modified-Since": "",
+				"Content-Encoding":  "",
+				"Cookie":            "",
+				"Max-Forwards":      "",
+			},
+			newRequestWithHeaders(map[string]string{
+				clients.ContentType: clients.ContentTypeJSON,
+				"If-Modified-Since": "If-Modified-Since",
+				"Content-Encoding":  "Content-Encoding",
+				"Cookie":            "Cookie",
+				"Max-Forwards":      "Max-Forwards",
+			}, http.MethodGet),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var loggerMock = logger.NewMockClient()
+			ctx := context.Background()
+			getCommand, _ := NewGetCommand(
+				testDevice,
+				testCommand,
+				ctx,
+				nil,
+				loggerMock,
+				tt.req)
+			actualHeaders := map[string]string{}
+			for _, h := range tt.requestedHeaders {
+				actualHeaders[h] = getCommand.(serviceCommand).Request.Header.Get(h)
+			}
```

Comment:
I think I was not clear in my last comment regarding testing all the headers that are present in the `getCommand` request. With this logic you are only verifying the `requestHeaders` contents from the `getCommand` which are present in `tt.requestedHeaders`. To ensure that not additional headers are present you can get all of the headers from the `getCommand` request and compare it to the tt.expectedHeaders. This will allow you to also remove the headers which have empty strings from the test arguments, since the test will now fail properly.

Please apply this comment to `put_test.go` as well.